Ruby build failing

I am unable to build the ruby gem. Here is the homebrew install:

```
$ curl -fsSL https://goo.gl/getgrpc | bash -s ruby
zlib is already installed
openssl is already installed
google-protobuf is already installed
grpc is already installed
installing grpc ruby
Building native extensions with: '--with-grpc-dir=/usr/local'
This could take a while...
ERROR:  Error installing grpc:
	ERROR: Failed to build gem native extension.

    /Users/blowmage/.rubies/ruby-2.2.3/bin/ruby -r ./siteconf20151005-44201-1btzn1u.rb extconf.rb --with-grpc-dir=/usr/local
checking for dlopen() in -ldl... yes
checking for inflate() in -lz... yes
checking for gpr_now() in -lgpr... yes
checking for grpc_channel_destroy() in -lgrpc... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling rb_byte_buffer.c
compiling rb_call.c
rb_call.c:177:32: error: too many arguments to function call, expected single argument 'call', have 2 arguments
  err = grpc_call_cancel(call, NULL);
        ~~~~~~~~~~~~~~~~       ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include/stddef.h:105:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
/usr/local/include/grpc/grpc.h:448:1: note: 'grpc_call_cancel' declared here
grpc_call_error grpc_call_cancel(grpc_call *call);
^
rb_call.c:192:10: error: implicit declaration of function 'grpc_call_get_peer' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  peer = grpc_call_get_peer(call);
         ^
rb_call.c:192:10: note: did you mean 'grpc_rb_call_get_peer'?
rb_call.c:187:14: note: 'grpc_rb_call_get_peer' declared here
static VALUE grpc_rb_call_get_peer(VALUE self) {
             ^
rb_call.c:192:8: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Werror,-Wint-conversion]
  peer = grpc_call_get_peer(call);
       ^ ~~~~~~~~~~~~~~~~~~~~~~~~
rb_call.c:560:25: error: no member named 'reserved' in 'struct grpc_op'
    st->ops[st->op_num].reserved = NULL;
    ~~~~~~~~~~~~~~~~~~~ ^
rb_call.c:655:70: error: too many arguments to function call, expected 4, have 5
  err = grpc_call_start_batch(call, st.ops, st.op_num, ROBJECT(tag), NULL);
        ~~~~~~~~~~~~~~~~~~~~~                                        ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include/stddef.h:105:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
/usr/local/include/grpc/grpc.h:420:1: note: 'grpc_call_start_batch' declared here
grpc_call_error grpc_call_start_batch(grpc_call *call, const grpc_op *ops,
^
5 errors generated.
make: *** [rb_call.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/blowmage/.gem/repos/grpc-travis/gems/grpc-0.11.0 for inspection.
Results logged to /Users/blowmage/.gem/repos/grpc-travis/extensions/x86_64-darwin-14/2.2.0-static/grpc-0.11.0/gem_make.out
```

And here is a `bundle install`:

```
$ bundle install
Fetching gem metadata from http://rubygems.org/..........
Fetching version metadata from http://rubygems.org/..
Resolving dependencies...
Using addressable 2.3.8
Using extlib 0.9.16
Using multipart-post 2.0.0
Using faraday 0.9.1
Using google-protobuf 3.0.0.alpha.4.0
Using jwt 1.5.1
Using little-plugger 1.1.4
Using multi_json 1.11.2
Using logging 2.0.0
Using memoist 0.12.0
Using signet 0.6.1
Using googleauth 0.4.2
Installing grpc 0.11.0 with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /Users/blowmage/.rubies/ruby-2.2.3/bin/ruby -r ./siteconf20151005-44363-wztxjm.rb extconf.rb
checking for dlopen() in -ldl... yes
checking for inflate() in -lz... yes
checking for gpr_now() in -lgpr... yes
checking for grpc_channel_destroy() in -lgrpc... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling rb_byte_buffer.c
compiling rb_call.c
rb_call.c:177:32: error: too many arguments to function call, expected single argument 'call', have 2 arguments
  err = grpc_call_cancel(call, NULL);
        ~~~~~~~~~~~~~~~~       ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include/stddef.h:105:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
/usr/local/include/grpc/grpc.h:448:1: note: 'grpc_call_cancel' declared here
grpc_call_error grpc_call_cancel(grpc_call *call);
^
rb_call.c:192:10: error: implicit declaration of function 'grpc_call_get_peer' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  peer = grpc_call_get_peer(call);
         ^
rb_call.c:192:10: note: did you mean 'grpc_rb_call_get_peer'?
rb_call.c:187:14: note: 'grpc_rb_call_get_peer' declared here
static VALUE grpc_rb_call_get_peer(VALUE self) {
             ^
rb_call.c:192:8: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Werror,-Wint-conversion]
  peer = grpc_call_get_peer(call);
       ^ ~~~~~~~~~~~~~~~~~~~~~~~~
rb_call.c:560:25: error: no member named 'reserved' in 'struct grpc_op'
    st->ops[st->op_num].reserved = NULL;
    ~~~~~~~~~~~~~~~~~~~ ^
rb_call.c:655:70: error: too many arguments to function call, expected 4, have 5
  err = grpc_call_start_batch(call, st.ops, st.op_num, ROBJECT(tag), NULL);
        ~~~~~~~~~~~~~~~~~~~~~                                        ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include/stddef.h:105:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
/usr/local/include/grpc/grpc.h:420:1: note: 'grpc_call_start_batch' declared here
grpc_call_error grpc_call_start_batch(grpc_call *call, const grpc_op *ops,
^
5 errors generated.
make: *** [rb_call.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/blowmage/.gem/repos/grpc-travis/gems/grpc-0.11.0 for inspection.
Results logged to /Users/blowmage/.gem/repos/grpc-travis/extensions/x86_64-darwin-14/2.2.0-static/grpc-0.11.0/gem_make.out
An error occurred while installing grpc (0.11.0), and Bundler cannot continue.
Make sure that `gem install grpc -v '0.11.0'` succeeds before bundling.
```

Another report was made at grpc/grpc-common#283.

I continue to get compiler errors when installing the ruby gem on a fresh OS X 10.11. The errors I get now are:

```
current directory: /Users/blowmage/.gem/repos/grpc/gems/grpc-0.12.0/src/ruby/ext/grpc
make "DESTDIR=" clean

current directory: /Users/blowmage/.gem/repos/grpc/gems/grpc-0.12.0/src/ruby/ext/grpc
make "DESTDIR="
compiling rb_byte_buffer.c
compiling rb_call.c
compiling rb_call_credentials.c
rb_call_credentials.c:79:61: error: unused parameter 'args' [-Werror,-Wunused-parameter]
static VALUE grpc_rb_call_credentials_callback_rescue(VALUE args,
                                                            ^
rb_call_credentials.c:134:59: error: unused parameter 'state' [-Werror,-Wunused-parameter]
static void grpc_rb_call_credentials_plugin_destroy(void *state) {
                                                          ^
2 errors generated.
make: *** [rb_call_credentials.o] Error 1

make failed, exit code 2
```

The [full log is here](https://gist.github.com/blowmage/c52e36a62f4134443b26).

--- a/src/ruby/ext/grpc/rb_call_credentials.c
+++ b/src/ruby/ext/grpc/rb_call_credentials.c
@@ -78,6 +78,7 @@ static VALUE grpc_rb_call_credentials_callback(VALUE callback_args) {
 
 static VALUE grpc_rb_call_credentials_callback_rescue(VALUE args,
                                                       VALUE exception_object) {
+  (void)args;
   VALUE result = rb_hash_new();
   rb_hash_aset(result, rb_str_new2("metadata"), Qnil);
   /* Currently only gives the exception class name. It should be possible get
@@ -132,6 +133,7 @@ static void grpc_rb_call_credentials_plugin_get_metadata(
 }
 
 static void grpc_rb_call_credentials_plugin_destroy(void *state) {
+  (void)state;
   // Not sure what needs to be done here
 }
 

--- a/src/ruby/ext/grpc/rb_call_credentials.c
+++ b/src/ruby/ext/grpc/rb_call_credentials.c
@@ -78,8 +78,8 @@ static VALUE grpc_rb_call_credentials_callback(VALUE callback_args) {
 
 static VALUE grpc_rb_call_credentials_callback_rescue(VALUE args,
                                                       VALUE exception_object) {
-  (void)args;
   VALUE result = rb_hash_new();
+  (void)args;
   rb_hash_aset(result, rb_str_new2("metadata"), Qnil);
   /* Currently only gives the exception class name. It should be possible get
      more details */

--- a/src/ruby/ext/grpc/rb_event_thread.c
+++ b/src/ruby/ext/grpc/rb_event_thread.c
@@ -102,6 +102,7 @@ static void grpc_rb_event_queue_destroy() {
 
 static void *grpc_rb_wait_for_event_no_gil(void *param) {
   grpc_rb_event *event = NULL;
+  (void)param;
   gpr_mu_lock(&event_queue.mu);
   while ((event = grpc_rb_event_queue_dequeue()) == NULL) {
     gpr_cv_wait(&event_queue.cv,
@@ -117,6 +118,7 @@ static void *grpc_rb_wait_for_event_no_gil(void *param) {
 }
 
 static void grpc_rb_event_unblocking_func(void *arg) {
+  (void)arg;
   gpr_mu_lock(&event_queue.mu);
   event_queue.abort = true;
   gpr_cv_signal(&event_queue.cv);
@@ -127,6 +129,7 @@ static void grpc_rb_event_unblocking_func(void *arg) {
  * events */
 static VALUE grpc_rb_event_thread(VALUE arg) {
   grpc_rb_event *event;
+  (void)arg;
   while(true) {
     event = (grpc_rb_event*)rb_thread_call_without_gvl(
         grpc_rb_wait_for_event_no_gil, NULL,

